index.rst

```diff
@@ -352,28 +385,145 @@ To make sure the system would work properly we had to enable two additional feat
+For instance, attempts to bring all salkafka producers up at the same time would sometimes cause daemons to fail and, sometimes, crash.
+On other occasions, it was possible to start one producer at a time but the final set of producers would experience the HRTO_ problem we saw before.
+On occasions where it was possible to standup the system successfully it would run for a certain number of hours and then one of the daemons would fail and terminate, killing all the components that where connected to it.
+The most common occurrences we noticed was that the daemon M1M3 component was attached to would fail, causing the component to crash.
+
+Investigating the log generated by the daemons, we concluded that once we transitioned to deploying an even larger system than before, the daemons themselves started to run out of resources.
+DDS contains some builtin topics used by the middleware system for self configuration and bookkeeping.
+When the problems started to manifest the daemons would log messages of missing heartbeats (this is the DDS internal heartbeat, used to determine the presence of a domain participant) and other missing builtin topics (e.g. ``d_sampleChain``), thus indicating that the daemons where having problems keeping up with the load;
+
+  .. code-block:: text
+
+    ========================================================================================
+    Report      : INFO
+    Date        : 2020-09-03T17:19:02+0000
+    Description : Missed heartbeat for node 105060556. (DCPSPublication is complete)
+    Node        : ospl-daemon-m482p
+    Process     : spliced <148347>
+    Thread      : OSPL Garbage Collector 7f48013ae700
+    Internals   : P674/6.10.4/de6011b/2a9a655/v_spliced/v_spliced.c/2034/769/1599153542.755942957/2
+    ========================================================================================
+    Report      : WARNING
+    Date        : 2020-09-03T17:40:49+0000
+    Description : writer 6672e859:7b:1:a02 topic d_sampleChain waiting on high watermark due to reader 4254ba5d:7f:1:1507
+    Node        : ospl-daemon-m482p
+    Process     : ddsi2 <148361>
+    Thread      : xmit.user 7f20e331b700
+    Internals   : P674/6.10.4/de6011b/2a9a655/ddsi2/q_transmit.c/923/0/1599154849.155224251/2
+    ========================================================================================
+
+In order to solve this issue, it required some further understanding of how DDS communication works and more study on the daemons configuration.
+The goal was to be able to increase the resources available for the OpenSplice daemons to handle the increased load while also reducing the load on the DDS services.
+
+.. _Increasing-OpenSplice-daemon-resources:
+
+Increasing OpenSplice daemon resources
+--------------------------------------
+
+The OpenSplice DDS software configuration provides a couple of different ways of increasing the computing resources available for the daemons, mostly revolving around the allotted memory.
```

Comment:
Please consider "a couple of" -> "several" and "the allotted memory" -> "memory allocation limits"